internal/pkg/describe/pipeline.go

```diff
@@ -0,0 +1,119 @@
+)
+
+type stackResourcesDescriber interface {
+	StackResources(stackName string) ([]*cloudformation.StackResource, error)
+}
+
+type pipelineGetter interface {
+	GetPipeline(pipelineName string) (*codepipeline.Pipeline, error)
+}
+
+// Project contains serialized parameters for a pipeline.
+type Pipeline struct {
+	codepipeline.Pipeline
+
+	Resources []*CfnResource `json:"resources,omitempty"`
+}
+
+// PipelineDescriber retrieves information about an application.
+type PipelineDescriber struct {
+	pipelineName  string
+	showResources bool
+
+	pipelineSvc             pipelineGetter
+	stackResourcesDescriber stackResourcesDescriber
+}
+
+// NewPipelineDescriber instantiates a new pipeline describer
+func NewPipelineDescriber(pipelineName string, showResources bool) (*PipelineDescriber, error) {
+	sess, err := session.NewProvider().Default()
+	if err != nil {
+		return nil, err
+	}
+
+	describer := newStackDescriber(sess)
+	pipelineSvc := codepipeline.New(sess)
+
+	return &PipelineDescriber{
+		pipelineName:            pipelineName,
+		pipelineSvc:             pipelineSvc,
+		showResources:           showResources,
+		stackResourcesDescriber: describer,
+	}, nil
+}
+
+func (d *PipelineDescriber) Describe() (HumanJSONStringer, error) {
+	cp, err := d.pipelineSvc.GetPipeline(d.pipelineName)
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline: %w", err)
+	}
+	var resources []*CfnResource
+	if d.showResources {
+		stackResources, err := d.stackResourcesDescriber.StackResources(d.pipelineName)
+		if err != nil && !IsStackNotExistsErr(err) {
+			return nil, fmt.Errorf("retrieve pipeline resources: %w", err)
+		}
+		resources = flattenResources(stackResources)
+	}
+	pipeline := &Pipeline{*cp, resources}
+	return pipeline, nil
+}
+
+// JSONString returns the stringified Pipeline struct with JSON format.
+func (p *Pipeline) JSONString() (string, error) {
+	b, err := json.Marshal(p)
+	if err != nil {
+		return "", fmt.Errorf("marshal pipeline: %w", err)
+	}
+	return fmt.Sprintf("%s\n", b), nil
+}
+
+// HumanString returns the stringified Pipeline struct with human readable format.
+func (p *Pipeline) HumanString() string {
+	var b bytes.Buffer
+	// TODO tweak the spacing
+	writer := tabwriter.NewWriter(&b, minCellWidth, tabWidth, cellPaddingWidth, paddingChar, noAdditionalFormatting)
+	fmt.Fprintf(writer, color.Bold.Sprint("About\n\n"))
+	writer.Flush()
+	fmt.Fprintf(writer, "  %s\t%s\n", "Name", p.Name)
+	fmt.Fprintf(writer, "  %s\t%s\n", "Region", p.Region)
+	fmt.Fprintf(writer, "  %s\t%s\n", "AccountID", p.AccountID)
+	fmt.Fprintf(writer, "  %s\t%s\n", "Created At", humanizeTime(time.Unix(p.CreatedAt.Unix(), 0)))
+	fmt.Fprintf(writer, "  %s\t%s\n", "Updated At", humanizeTime(time.Unix(p.UpdatedAt.Unix(), 0)))
+	writer.Flush()
+	fmt.Fprintf(writer, color.Bold.Sprint("\nStages\n\n"))
+	writer.Flush()
+	fmt.Fprintf(writer, "  %s\t%s\t%s\t%s\n", "Name", "Category", "Provider", "Details")
+	writer.Flush()
```

Comment:
it doesn't hurt but this flush might be redundant. Ideally there should be only one flush but we call flush after each place where color package get used in order to keep the format consistent.